tests/bluetooth/tester/src/gatt.c

```diff
@@ -1657,8 +1658,10 @@ static void write_long(uint8_t *data, uint16_t len)
 
 static struct bt_gatt_subscribe_params subscribe_params;
 
-/* ev header + default MTU_ATT-3 */
-static uint8_t ev_buf[33];
+/* TODO there should be better way of determining max supported MTU */
+#define MAX_NOTIF_DATA (MIN(BT_L2CAP_RX_MTU, BT_L2CAP_TX_MTU) - 3)
```

Comment:
This is basically doing `BT_ATT_MTU - sizeof(bt_att_hdr) - sizeof(bt_att_notify)`, so perhaps those values and structs could be used instead

Reply:
yes, but BT_ATT_MTU is defined in subsys/bluetooth/host/att_internal.h

maybe we should have BT_ATT_MTU_MAX or something in public headers?

Reply:
We could do that. Is there an issue with using internal headers in the tester? Alternatively a "tester header"? 

Personally I don't see any issues with using internal information in the tester, as a tester would sometimes needs information/access beyond what is in the public API.

Reply:
nah, this is just something that pop up while working on tester (ie it may be useful for other applications too)
I think we can leave it as is in tester for now

Reply:
Come to think of it, to handle long notifications, the application does need to know not only the ATT MTU (not `BT_ATT_MTU`, but the current ATT MTU that was exhanged) as well as the -3 (opcode + handle) to determine if the notification was (possibly) truncated.

Reply:
for current MTU on connection there is uint16_t bt_gatt_get_mtu(struct bt_conn *conn);

Reply:
> for current MTU on connection there is uint16_t bt_gatt_get_mtu(struct bt_conn *conn);

Correct, but the `struct bt_att_hdr` and `struct bt_att_notify` are internal, so the application will need to know that the first 3 octets is omitted and why, e.g. the MTU may be 30, but a data length of 27 may indicate that the notification perhaps was truncated, and the client should do a read long procedure. 

Anyways, that is outside the scope of this PR.

Reply:
@sjanc Did you have any plans on updating this PR to use the sizeof of the structs?

Reply:
no (at least not in that PR) since those are not defined in public headers (ie not intended to be used by applications)

Reply:
OK. I guess the best solution to this is then to create a GH issue requesting a way to determine this. As mentioned, these values may be useful for applications. 

Will you create a GH issue for it?

Reply:
enhancement issue created https://github.com/zephyrproject-rtos/zephyr/issues/43207